contents/blog/building-a-new-feature-e2e.md

```diff
@@ -0,0 +1,81 @@
+---
+date: 2023-03-17
+title: Building a new feature end-to-end
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+author:
+  - yakko-majuri
+category: Engineering
+tags:
+  - Guides
+  - Product engineers
+---
+
+Having spent most of my time during the past few years working deep in our backend, I recently found myself building out a new client-facing feature ([sampling](/manual/sampling)) end-to-end.
+
+Doing so gave me a renewed perspective on what a the process for shipping new features should look like, so I thought I'd share a bit about how things went. Hopefully I can provide some insight on how to approach building something new and seeing it all the way through to the end - having it live and with usage!
```

Comment:
Would you say this is normally how building a new feature is at PostHog? If yes, then maybe we can frame this as "How we build a new feature at PostHog". Conclusion could talk about how this is different than other development processes (shipping fast, autonomy to make your own decisions, testing in production).

If not, then this could be "how we aspire to ship a new feature at PostHog" and potentially talk about gaps that are preventing this from being the standard.

Reply:
"How product engineers build new features" or "How a product engineer builds a new feature"

Reply:
After reading the first draft, I'd be inclined to steer away from "product engineer" headlines and simply go with something like "How we built a data sampling MVP in x weeks/days". We can weave the product engineer theme into the article without overtly pointing at it in the headline.